src/Symfony/Component/Console/Style/SymfonyStyle.php

```diff
@@ -388,6 +394,22 @@ public function newLine($count = 1)
     }
 
     /**
+     * Sets the input stream used by the SymfonyQuestionHelper.
+     *
+     * @param resource $stream
+     *
+     * @throws InvalidArgumentException If the given stream is not a resource
+     */
+    public function setInputStream($stream)
+    {
+        if (!is_resource($stream)) {
+            throw new InvalidArgumentException(sprintf('Input stream must be a valid resource, %s given', gettype($stream)));
```

Comment:
Imo you should simply remove this check. The docblock already states that the method expects a resource (we never have checks like this).